spec.emu

```diff
@@ -1,23 +1,333 @@
+
+    <emu-table id="table-duration-components">
+      <emu-caption>Components of Duration Instances</emu-caption>
+      <table class="real-table">
+        <thead>
+          <tr>
+            <th>Internal Slot</th>
+            <th>Property</th>
+          </tr>
+        </thead>
+        <tr>
+          <td>[[Years]]</td>
+          <td>`"year"`</td>
+        </tr>
+        <tr>
+          <td>[[Months]]</td>
+          <td>`"months"`</td>
+        </tr>
+        <tr>
+          <td>[[Weeks]]</td>
+          <td>`"weeks"`</td>
+        </tr>
+        <tr>
+          <td>[[Days]]</td>
+          <td>`"days"`</td>
+        </tr>
+        <tr>
+          <td>[[Hours]]</td>
+          <td>`"hours"`</td>
+        </tr>
+        <tr>
+          <td>[[Minutes]]</td>
+          <td>`"minutes"`</td>
+        </tr>
+        <tr>
+          <td>[[Seconds]]</td>
+          <td>`"seconds"`</td>
+        </tr>
+        <tr>
+          <td>[[Milliseconds]]</td>
+          <td>`"milliseconds"`</td>
+        </tr>
+        <tr>
+          <td>[[Microseconds]]</td>
+          <td>`"microseconds"`</td>
+        </tr>
+        <tr>
+          <td>[[Nanoseconds]]</td>
+          <td>`"nanonseconds"`</td>
+        </tr>
+      </table>
+    </emu-table>
+
+    <emu-note type=editor>The [[Weeks]] field is still under discussion in Temporal. For more, check out the <a href="https://github.com/tc39/proposal-temporal/issues/532">Github thread</a> on the Temporal issue tracker.</emu-note>
+
+    <emu-clause id="sec-partitiondurationformatpattern" aoid="PartitionDurationFormatPattern">
+      <h1>PartitionDurationFormatPattern ( _durationFormat_, _duration_ )</h1>
+
+      <p>The PartitionDurationFormatPattern abstract operation is called with the arguments _durationFormat_ (which must be an object initialized as a DurationFormat) and _duration_ (which must be a Record with the fields enumerated in <emu-xref href="#table-duration-components"></emu-xref> ), and creates the corresponding parts according to the effective locale and the formatting options of _durationFormat_. The following steps are taken:</p>
+
+      <emu-alg>
+        1. Let _result_ be a new empty List.
+        1. If _durationFormat_.[[Style]] is `"dotted"`, then
+          1. Let _patternParts_ be PartitionPattern(_durationFormat_.[[Templates]]).
+          1. For each _patternPart_ in _patternParts_, do
+            1. Let _p_ be _patternPart_.[[Type]].
+            1. If _p_ is `"literal"`, then
+              1. Append _patternPart_ to the end of _result_.
+            1. Else if _duration_[[&lt;_p_&gt;]] is not *undefined* and _p_ exists in _durationFormat_.[[Fields]], then
+              1. Let _num_ be FormatNumeric(_durationFormat_.[[NumberFormat]], _duration_[[&lt;_p_&gt;]]).
+              1. Append the new Record { [[Type]]: _p_, [[Value]]: _num_ } to the end of _result_.
```

Comment:
I think we should include the individual numeric parts here, not just lump it all together. You can do this by calling PartitionNumberPattern rather than FormatNumeric, as RelativeTimeFormat does.